packages/driver/cypress/integration/commands/navigation_spec.js

```diff
@@ -1441,9 +1410,23 @@ describe('src/cy/commands/navigation', () => {
         cy.on('fail', (err) => {
           const { lastLog } = this
 
-          expect(err.message).to.include('`cy.visit()` failed because you are attempting to visit a URL that is of a different origin.')
-          expect(err.message).to.include('The new URL is considered a different origin because the following parts of the URL are different:')
-          expect(err.message).to.include('> port')
+          expect(err.message).to.equal(stripIndent`\
+          \`cy.visit()\` failed because you are attempting to visit a URL that is of a different origin.\n
+          You likely forgot to use \`cy.origin()\`:\n
+          \`cy.visit('http://localhost:3500/fixtures/generic.html')\`
+          \`<other commands targeting http://localhost:3500 go here>\`\n
+          \`cy.origin('http://localhost:3501', () => {\`
+          \`  cy.visit('http://localhost:3501/fixtures/generic.html')\`
+          \`  <other commands targeting http://localhost:3501 go here>\`
+          \`})\`\n
+          The new URL is considered a different origin because the following parts of the URL are different:\n
+            > port\n
+          You may only \`cy.visit()\` same-origin URLs within a single test.\n
+          The previous URL you visited was:\n
+            > 'http://localhost:3500'\n
+          You're attempting to visit this URL:\n
+            > 'http://localhost:3501'`)
```

Comment:
This feels brittle. Any reason we can't pull this error message in and check against that?

Reply:
I think a little brittleness is better than the downsides of pulling in the error message. If we do that, the test becomes less readable, because it would require looking up the error message in order a different file to see what's being asserted. It also would reduce test coverage since this covers dynamic values being applied to the error message. A typo in the error message may be caught when writing a test like this, whereas it would be missed if pulling in the error message.